hummingbird/ml/_topology.py

```diff
@@ -372,10 +373,21 @@ def fix_graph(graph):
             container = PyTorchSklearnContainerClassification
 
     n_threads = None if constants.N_THREADS not in extra_config else extra_config[constants.N_THREADS]
-    batch_size = None if constants.BATCH_SIZE not in extra_config else extra_config[constants.BATCH_SIZE]
-    hb_model = container(hb_model, n_threads, batch_size, extra_config=extra_config)
-
-    return hb_model
+    batch_size = None if constants.TEST_INPUT not in extra_config else _get_batch_size(test_input)
+    hb_container = container(hb_model, n_threads, batch_size, extra_config=extra_config)
+
+    if remainder_model:
+        aux_container = container(remainder_model, n_threads, remainder_size, extra_config=extra_config)
+        return BatchContainer(hb_container, aux_container)
+    elif remainder_size is not None and remainder_size > 0:
+        # torch backend case
+        aux_container = container(hb_model, n_threads, remainder_size, extra_config=extra_config)
+        return BatchContainer(hb_container, aux_container)
+    elif remainder_size is not None:
+        assert remainder_size == 0, "remainder_size is non zero but no remainder_model has been created"
```

Comment:
What is this case covering? When `remainder_size` is equal to `batch_size`?

Reply:
No, this is for the case when `remainder_size` is not `None` and remainder_size == 0. I make sure that `remainder_size` is not `None` only if `convert_batch` is called, see

https://github.com/microsoft/hummingbird/blob/2eda7538106888d56b6a3b1d1a70c0d33fa15354/hummingbird/ml/convert.py#L406
https://github.com/microsoft/hummingbird/blob/2eda7538106888d56b6a3b1d1a70c0d33fa15354/hummingbird/ml/convert.py#L414

In the case `remainder_size` is 0, no `remainder_model` is compiled. But since a user has asked for `BatchContainer` using `convert_batch`, we return `BatchContainer` but without the container for `remainder_model`.